### PR TITLE
feat: cleanup docs repository

### PR DESCRIPTION
**Pull request by NucciTheBoss**
This PR blows out some of the files that we don't need to have within the documentation repository such as the cheat sheets and initialization scripts, and adds a project specific README, contributing guidelines, and CC-BY-SA license file.

Also adds a `.editorconfig` file that sets the default indentation for markdown files to 2 spaces. We should also eventually look into using Terraform to set in place the issue labels that I had originally for better issue tagging. Mostly just a chore to go through a set up by hand :sweat_smile:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+* `Status: Help Wanted` - Issues where we need help from the Ubuntu HPC community to solve.
+
+For a complete look at Charmed HPC's documentation labels, see the
+[project labels page](https://github.com/charmed-hpc/docs/labels).
+
+## Bug Reports
+
+A bug is a *demonstrable problem* that is caused by errors in Charmed HPC's
+documentation. Good bug reports make Charmed HPC's documentation better, so
+thankyou for taking the time to report issues!
```
Comment (jedel1043) [inline CONTRIBUTING.md:77]:
```suggestion
thank you for taking the time to report issues!
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+   $ git clone git@github.com:<your-username>/docs.git
+
+   # Navigate to the newly cloned directory
+   $ cd docs
+
+   # Assign the original repo to a remote called "upstream"
+   $ git remote add upstream git@github.com:charmed-hpc/docs.git
+   ```
+
+2. If you cloned a the documentation a while ago, pull the latest changes from the
```
Comment (jedel1043) [inline CONTRIBUTING.md:148]:
```suggestion
2. If you cloned the documentation a while ago, pull the latest changes from the
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+   ```shell
+   git push origin <topic-branch-name>
+   ```
+
+8. [Open a Pull Request](https://help.github.com/articles/about-pull-requests/)
+    with a clear title and description against the `main` branch.
+
+> By submitting your changes to Charmed HPC's documentation, you agree to license your
+> contributions under the terms of the [Creative Commons Attribution-Share Alike
+> International license, version 4.0](./LICENSE).
```
Comment (jedel1043) [inline CONTRIBUTING.md:207]: Is this necessary? There's already a license section below.

**Review by jedel1043 — COMMENTED** [review]
Nice!

File: CONTRIBUTING.md (hunk 2 above)
Comment (NucciTheBoss) [reply CONTRIBUTING.md:148]: Oof - `codespell` needs a complimenting `grammarcheck`

File: CONTRIBUTING.md (hunk 3 above)
Comment (NucciTheBoss) [reply CONTRIBUTING.md:207]: Eh, not really, it's just something that I've carried from the original CONTRIBUTING.md template I put together like 5 years ago :skull: 

I think it we can be DRY here as we're effectively repeating the same blurb later within the document.

```suggestion
> By submitting your changes to Charmed HPC's documentation, you agree to license your
> contributions under the terms of the [Creative Commons Attribution-Share Alike
> International license, version 4.0](./LICENSE).
```

**Comment by NucciTheBoss** [comment]
@jedel1043 applied your suggested doc fixes. Thank you for catching them! Get some credit towards our "peas in a pod" GitHub achievement :pea_pod:

**Review by jedel1043 — APPROVED** [review]
Looks great!

**Pull request merged**